packages/aws-cdk-lib/aws-apigateway/lib/deployment.ts

```diff
@@ -137,6 +151,7 @@ class LatestDeploymentResource extends CfnDeployment {
     super(scope, id, {
       description: props.description,
       restApiId: props.restApi.restApiId,
+      stageName: props.stageName,
```

Comment:
Could you please elaborate why we need to add the `stageName` property to LatestDeployment? Based on my previous comment, I thought inorder to use Deployment, restApi will be created with `Deploy:false` which will not automatically create the deployment and stage. Hence in that case latestDeployment will not be used if `deploy:false` is set ([See here](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_apigateway.RestApi.html#latestdeployment)).

Reply:
When the user manually creates a `Deployment` the constructor creates a new `LatestDeploymentResource` (see [here](https://github.com/aws/aws-cdk/blob/main/packages/aws-cdk-lib/aws-apigateway/lib/deployment.ts#L71)), so that's why it's getting added here